wolfcrypt/src/sha256.c

```diff
@@ -582,8 +582,13 @@ static int InitSha256(wc_Sha256* sha256)
         0x90BEFFFAL, 0xA4506CEBL, 0xBEF9A3F7L, 0xC67178F2L
     };
 
+#ifdef WOLFSSL_SHA256_BY_SPEC
     #define Ch(x,y,z)       ((z) ^ ((x) & ((y) ^ (z))))
     #define Maj(x,y,z)      ((((x) | (y)) & (z)) | ((x) & (y)))
+#else
+    #define Ch(x,y,z)       ((((y) ^ (z)) & (x)) ^ (z))
```

Comment:
What's the speed difference?  Should we have the old way be the default until more testing?

Reply:
I changed it so the default Ch/Maj math based on specification is enabled. No speed difference on Intel platforms. Its mainly a compiler difference.

For the `SHA256_MANY_REGISTERS` it is slower, but can be faster on some platforms depending on the number of registers available. The `SHA256_MANY_REGISTERS` puts all data into registers for processing. On Intel this was 13% slower.